### PR TITLE
used <fieldset> and <legend> tag because of the semantics

### PR DESCRIPTION
**Pull request by NoahLiechti**
Changes proposed in this pull request:

 - Implemented `<fieldset>` (tag is used to group related elements in a form) 
 - Implemented `<legend>` instead of `<span> `(tag is used to write a caption)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
 
 # How Has This Been Tested?
 
local
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my own code
 - [ ] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/m-form-group/_template.js [hunk 1]
```diff
@@ -11,19 +11,21 @@ export default function ({
 
   return [
     html`
-      <label class="m-form-group__wrapper js-form-group__wrapper"> 
-        <span class="m-form-group__label-icon-wrapper js-form-group__label-icon-wrapper">
-          <span class="m-form-group__label">
-            ${label}
+      <label class="m-form-group__wrapper js-form-group__wrapper">
+        <fieldset>
```
Comment (AndyOGo) [inline src/components/m-form-group/_template.js]: `<fieldset>` within `<label>` is semantically and technically incorrect.
[`<label>`](https://www.w3.org/TR/2011/WD-html5-author-20110705/the-label-element.html) are only allowed to container [phrasing content](https://www.w3.org/TR/2011/WD-html5-author-20110705/content-models.html#phrasing-content).

File: src/components/m-form-group/_template.js [hunk 2]
```diff
@@ -11,19 +11,21 @@ export default function ({
...
+          <span class="m-form-group__label-icon-wrapper js-form-group__label-icon-wrapper">
```
Comment (LucaMele) [inline src/components/m-form-group/_template.js]: here as well maybe span is not needed

**Comment by NoahLiechti** [comment]
Oke I'll have a look at the HTML Spec

**Comment by NoahLiechti** [comment]
could you please review it one more time? Thanks!

File: src/components/m-form-group/_template.js [hunk 3]
```diff
@@ -11,20 +11,20 @@ export default function ({
 
   return [
     html`
-      <label class="m-form-group__wrapper js-form-group__wrapper"> 
-        <span class="m-form-group__label-icon-wrapper js-form-group__label-icon-wrapper">
-          <span class="m-form-group__label">
+      <fieldset class="m-form-group__wrapper js-form-group__wrapper">
```
Comment (AndyOGo) [inline src/components/m-form-group/_template.js]: If the `<legend>` below has a class element suffix of `__legend`, I would do the same for the `<fieldset>` replacing `wrapper` by `filedset` 🤔

File: src/components/m-form-group/_preview.html [hunk 4]
```diff
@@ -34,6 +34,11 @@
                   info="Caramels liquorice ice cream gingerbread gummi bears. Lemon drops cookie wafer cake. Sweet roll sesame snaps chocolate bar bear claw lemon drops sugar plum cupcake candy canes icing.">
     <axa-input value="Pre-filled content" name="test-name" error="Some error!"></axa-input>
   </axa-form-group>
+  <axa-form-group label="Test Label"
```
Comment (AndyOGo) [inline src/components/m-form-group/_preview.html]: we could discuss to rename `label` to `legend` for consistency, though this would be breaking change 🤔

**Pull request merged**